src/Agent.Plugins/PipelineArtifact/Telemetry/FileShareActionRecord.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using Agent.Sdk;
+using Microsoft.VisualStudio.Services.Content.Common.Telemetry;
+
+namespace Agent.Plugins.PipelineArtifact.Telemetry
+{
+    /// <summary>
+    /// Generic telemetry record for use with FileShare Artifact events.
+    /// </summary>
+    public class FileShareActionRecord : PipelineArtifactActionRecord
+    {
+        public int TotalFile { get; private set; }
+        public long TotalContentSize { get; private set; }
+        public long TimeLapse { get; private set; }
+        public string Command { get; private set; }
+        public int ExitCode { get; private set; }
+        public IList<ArtifactRecord> ArtifactRecords { get; private set; }
+
+        public FileShareActionRecord(TelemetryInformationLevel level, Uri baseAddress, string eventNamePrefix, string eventNameSuffix, AgentTaskPluginExecutionContext context, uint attemptNumber = 1)
+            : base(level, baseAddress, eventNamePrefix, eventNameSuffix, context, attemptNumber)
+        {
+        }
+
+        protected override void SetMeasuredActionResult<T>(T value)
+        {
+            if (value is FileSharePublishResult)
+            {
+                FileSharePublishResult result = value as FileSharePublishResult;
+                Command = result.Command;
+                ExitCode = result.ExitCode;
+            }
+
+            if (value is FileShareDownloadResult)
+            {
+                FileShareDownloadResult result = value as FileShareDownloadResult;
+                TotalFile = result.FileCount;
+                TotalContentSize = result.ContentSize;
+                this.ArtifactRecords = result.ArtifactRecords;
+            }
```

Comment:
else throw not supported exception, someone coded something incorrectly here

Reply:
I don't think it's necessary here? The PipelineArtifactActionRecord doesn't have this check either and Telemetry shouldn't make the task to fail.